### PR TITLE
magic-attach: Add another failure check

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
magic-attach: Add another failure check

Add exception handling for the situation where the magic-attach endpoint is unavailable

## Test Steps
Run the new unit tests for this feature

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by CalvoM — APPROVED** [review]
LGTM

**Review by orndorffgrant — COMMENTED** [review]
Looks good but per https://chat.canonical.com/canonical/pl/sqyk8koq7iyhtxihjoarqozdbc I thought we were also going to increase the poll interval in `wait` when we get this code

**Review by alnvdl-work — APPROVED** [review]

**Comment by lucasmoura** [comment]
@orndorffgrant updated

File: uaclient/api/tests/test_api_u_pro_attach_magic_wait_v1.py [hunk 1]
```diff
@@ -106,3 +106,42 @@ def test_wait_succeeds_after_number_of_connectivity_errors(
         assert expected_response.contract_id == "cid"
         assert 4 == m_attach_token_info.call_count
         assert 3 == m_sleep.call_count
+
+    @mock.patch("time.sleep")
+    def test_wait_succeeds_after_unavailable_server(
```
Comment (orndorffgrant) [inline uaclient/api/tests/test_api_u_pro_attach_magic_wait_v1.py:111]: a great test!

**Review by orndorffgrant — APPROVED** [review]
Looks good!
the test failure is unrelated. I'll fix in a separate PR

**Pull request merged**